This patch is revised version from openembed. It build host native binary
rather than target binary in cross-compile environment.

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>

Upstream-Status: Inappropriate [embedded specific]

Index: metacity-2.30.3/src/Makefile.am
===================================================================
--- metacity-2.30.3.orig/src/Makefile.am	2010-09-05 00:09:52.000000000 +0800
+++ metacity-2.30.3/src/Makefile.am	2010-12-31 11:15:31.000000000 +0800
@@ -134,11 +134,9 @@
 metacity_theme_viewer_SOURCES=  \
 	ui/theme-viewer.c
 
-schema_bindings_SOURCES =       \
-        core/schema-bindings.c			\
-	metacity.schemas.in.in
+schema_bindings:
+	@CC_FOR_BUILD@ core/schema-bindings.c -I./include -I../ @CFLAGS_FOR_BUILD@ @LDFLAGS_FOR_BUILD@ -o schema_bindings
 
-schema_bindings_LDADD = @METACITY_LIBS@
 metacity.schemas.in: schema_bindings ${srcdir}/metacity.schemas.in.in
 	@echo Generating keybinding schemas... ${srcdir}/metacity.schemas.in.in
 	${builddir}/schema_bindings ${srcdir}/metacity.schemas.in.in ${builddir}/metacity.schemas.in
Index: metacity-2.30.3/configure.in
===================================================================
--- metacity-2.30.3.orig/configure.in	2010-09-22 22:14:06.000000000 +0800
+++ metacity-2.30.3/configure.in	2010-12-31 11:15:31.000000000 +0800
@@ -31,6 +31,32 @@
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
+if test x"$CC_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+	else
+	       CC_FOR_BUILD="$CC"
+	fi
+fi
+AC_SUBST([CC_FOR_BUILD])
+if test x"$CFLAGS_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       echo boo!
+	else
+	       CFLAGS_FOR_BUILD="$CFLAGS"
+	fi
+fi
+AC_SUBST([CFLAGS_FOR_BUILD])
+if test x"$LDFLAGS_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       echo boo!
+	else
+	       LDFLAGS_FOR_BUILD="$LDFLAGS"
+	fi
+fi
+AC_SUBST([LDFLAGS_FOR_BUILD])
+
+
 #### Integer sizes 
 
 AC_CHECK_SIZEOF(char)
